roles/reproducer/tasks/devscripts_layout.yml

```diff
@@ -1,6 +1,224 @@
 ---
-- name: Run devscripts role
-  vars:
-    cifmw_use_reproducer: true
+- name: Ensure we have up-to-date devscripts config
+  ansible.builtin.import_role:
+    name: "devscripts"
+    tasks_from: "build_config.yml"
+
+- name: Check if devscript cluster exists
   ansible.builtin.import_role:
-    name: devscripts
+    name: "devscripts"
+    tasks_from: "110_check_ocp.yml"
+
+- name: Push OCP related DNS entries
+  become: true
+  notify: "Restart dnsmasq"
+  vars:
+    _cluster: "{{ cifmw_devscripts_config['cluster_name'] }}"
+    _pub_net_data: >-
+      {{
+        _libvirt_manager_networking.networks[cifmw_libvirt_manager_pub_net]
+      }}
+    _api_addr: >-
+      {% if _pub_net_data.network_v4 is defined -%}
+      {{
+        _pub_net_data.network_v4 | ansible.utils.nthhost(2)
+      }}
+      {% endif -%}
+      {% if _pub_net_data.network_v6 is defined -%}
+      {{
+        _pub_net_data.network_v6 | ansible.utils.nthhost(2)
+      }}
+      {% endif -%}
+    _apps_addr: >-
+      {% if _pub_net_data.network_v4 is defined -%}
+      {{
+        _pub_net_data.network_v4 | ansible.utils.nthhost(3)
+      }}
+      {% endif -%}
+      {% if _pub_net_data.network_v6 is defined -%}
+      {{
+        _pub_net_data.network_v6 | ansible.utils.nthhost(3)
+      }}
+      {% endif -%}
+  ansible.builtin.copy:
+    dest: "/etc/cifmw-dnsmasq.d/ocp_dns.conf"
+    mode: "0644"
+    content: |
+      local=/{{ _cluster }}.openstack.lab/
+      domain={{ _cluster }}.openstack.lab
+      address=/api.{{ _cluster }}.openstack.lab/{{ _api_addr }}
+      address=/api-int.{{ _cluster }}.openstack.lab/{{ _api_addr }}
+      address=/.apps.{{ _cluster }}.openstack.lab/{{ _apps_addr }}
+      localise-queries
+    validate: "/usr/sbin/dnsmasq -C %s --test"
+
+- name: Configure NetworkManager
+  become: true
+  notify: Restart NetworkManager
+  ansible.builtin.copy:
+    dest: >-
+      /etc/NetworkManager/conf.d/{{ cifmw_reproducer_networkmanager_config }}
+    mode: "0644"
+    content: |
+      [main]
+      rc-manager=none
+      dns=none
+    validate: "NetworkManager --print-config %s"
+
+- name: Ensure services are reloaded
+  ansible.builtin.meta: flush_handlers
+
+- name: Ensure we use dnsmasq as resolver
```

Comment:
So after changing `gw_v4` to `gw_v6` my server lost the ability to resolve names.
The reason for this is that ocppr is configured with an IPv4 address `172.22.0.1`.

```
[zuul@host26 ~]$ cat /etc/resolv.conf
search ocp.openstack.lab
nameserver fd00:aaaa:0000:0000:0000:0000:0000:0001
[zuul@host26 ~]$ ping google.com
ping: google.com: Name or service not known
```

```
[zuul@host26 ~]$ sudo ss -natulp | grep 53
udp   UNCONN 0      0         172.22.0.1:53         0.0.0.0:*    users:(("dnsmasq",pid=74001,fd=7))     
tcp   LISTEN 0      32        172.22.0.1:53         0.0.0.0:*    users:(("dnsmasq",pid=74001,fd=8))  
```

I think we need to change the approach - we should leave NetworkManager's dns config as is. Detect if it is using dnsmasq or resolvd - and then add forwarding via the cifmw-dnsmasq for the internal domains only.